include/lib/allocation_tracker.hpp

```diff
@@ -60,13 +63,22 @@ class AllocationTracker {
   static inline bool is_active();
 
 private:
+  using AdressSet = std::unordered_set<uintptr_t>;
+
   struct TrackerState {
+    void init(bool track_alloc, bool track_dealloc) {
```

Comment:
not very idiomatic... Should I rely on constructor ? Can we afford to re-init the mutex.